packages/env-options/src/environment-options.js

```diff
@@ -85,7 +91,7 @@ export const makeEnvironmentCaptor = aGlobal => {
       const globalEnv = globalProcess.env;
       if (globalEnv && typeof globalEnv === 'object') {
         if (optionName in globalEnv) {
-          arrayPush(capturedEnvironmentOptionNames, optionName);
+          capturedEnvironmentOptionNames.push(optionName);
```

Comment:
This and the above `Fail` introduce late property access on built-ins; is that acceptable?

Reply:
@kriskowal , what do you think? Leaving this conversation open waiting for your answer.

Reply:
There is a layering problem. We impose mutually exclusive requirements of libraries that operate above, within, and below SES in the layers of the dependency graph. This change as written uses `env-options` as both within and above SES. That is, it’s used within SES and also above SES.

We currently require anything within SES to defend itself from mutations to the global environment that may occur in the start compartment between the initialization of SES and execution of lockdown. This is not strictly necessary, but we have maintained that high bar within SES using the `no-polymorphic-call` eslint rule. By taking `env-options` as a dependency of `ses`, it must meet that bar.

In accordance with being below SES, this code also rightly duplicates `Fail`, which it would not need to do if it were a library that assumes the pre-arrangement of a SES environment.

I appreciate avoiding duplication in source, but as a practical matter, the SES shim gets bundled in some environments, so this library would necessarily be duplicated in production regardless. That somewhat weakens the case for having a single shared implementation between SES internals and for use in SES environments. Not entirely, though.

In any case, acceptable outcomes in my opinion are:
1. `env-options` is largely duplicative of similar code in `ses` but not held to the `no-polymorphic-call` rule.
2. `env-options` continues to be used both within and above SES, in which case it must include `no-polymorphic-call` in its `eslint` config.

Reply:
Done